.travis.yml

```diff
@@ -250,28 +285,50 @@ jobs:
       before_install:
         - sudo bash ops/travis/system-testing/build_and_run before_install
       install:
-        - bash ops/travis/system-testing/build_and_run install
+        - unbuffer bash ops/travis/system-testing/build_and_run install
       before_script:
-        - bash ops/travis/system-testing/build_and_run before_script
+        - unbuffer bash ops/travis/system-testing/build_and_run before_script
       script:
-        - bash ops/travis/system-testing/build_and_run script
+        - unbuffer bash ops/travis/system-testing/build_and_run script
       after_success:
-        - bash ops/travis/system-testing/build_and_run after_success
+        - unbuffer bash ops/travis/system-testing/build_and_run after_success
       after_failure:
-        - bash ops/travis/system-testing/build_and_run after_failure
+        - unbuffer bash ops/travis/system-testing/build_and_run after_failure
+
+    - stage: deployment
+      name: master
+      if: branch = master
+      git:
+        depth: false
+      script: echo "Deploy master"
       deploy:
         - provider: script
-          skip_cleanup: true
-          script: bash ops/travis/staging/deploy
+          script: bash ops/travis/deploy/master
           on:
             branch: master
+    - stage: deployment
+      name: staging
+      if: branch = staging
+      git:
+        depth: false
+      script: echo "Deploy staging"
+      deploy:
         - provider: script
-          skip_cleanup: true
-          script: ops/travis/production/deploy
+          script: bash ops/travis/deploy/staging
           on:
-            branch: master
+            branch: staging
+    - stage: deployment
+      name: production/release
+      git:
+        depth: false
+      script: echo "Deploy production version $TRAVIS_TAG"
+      deploy:
+        - provider: script
+          script: ops/travis/deploy/production
+          on:
+            all_branches: true
```

Comment:
Does this mean that one can potentially release from any branch, not only stage, right ??
A use case would be a hot fix on a old commit from the stating branch

Reply:
when you do a git tag, this triggers a build on travis with the name of the tag. Basically that means it can be any name. On the next line you can see a condition that uses regexp to check that the tag follows semantic versioning.